sys/riscv/include/atomic.h

```diff
@@ -322,13 +324,82 @@ atomic_store_rel_64(volatile uint64_t *p, uint64_t val)
 #define	atomic_set_acq_long		atomic_set_acq_64
 #define	atomic_subtract_acq_long	atomic_subtract_acq_64
 
+#ifndef __CHERI_PURE_CAPABILITY__
 #define	atomic_add_acq_ptr		atomic_add_acq_64
 #define	atomic_clear_acq_ptr		atomic_add_acq_64
 #define	atomic_cmpset_acq_ptr		atomic_cmpset_acq_64
 #define	atomic_fcmpset_acq_ptr		atomic_fcmpset_acq_64
 #define	atomic_load_acq_ptr		atomic_load_acq_64
 #define	atomic_set_acq_ptr		atomic_set_acq_64
 #define	atomic_subtract_acq_ptr		atomic_subtract_acq_64
+#endif
+
+#ifdef __CHERI_PURE_CAPABILITY__
+static __inline void
+atomic_add_ptr(volatile uintptr_t *p, uintptr_t val)
+{
+
+	(void)__atomic_add_fetch(p, val, __ATOMIC_RELAXED);
```

Comment:
These won't work (yet) since they result in libcalls.

Reply:
Yes, the ones that the kernel actually uses I ended up patching to use inline implementations.

Reply:
This is apparently unused, so perhaps we leave it like this until we either will need an implementation or have compiler support?

Reply:
Sounds fine to me.